### PR TITLE
implemented favorite companies

### PR DESCRIPTION
**Pull request by jhayes45**
Students now have an array of favorites, type ObjectID.

Client can use the favorites array to check if individual companies are favorites.

Alternatively, if you want the full info for all favorite companies, we have a new route
GET /users/auth/favorites - students only, returns full employer objects of their favorite companies.

*After spring break:
Added another new route
PATCH /users/auth/favorites - students only, usage:

Request an employer_id (string) to be toggled as a favorite.
If it is a favorite, it will be removed.
If it is not a favorite, it will be added.

The response you get will be the updated array of employer_ids, called "favorites".

### CODE REVIEW

**Comment by jhayes45** [comment]
*After spring break:
Added another new route
PATCH /users/auth/favorites - students only, usage:

Request an employer_id (string) to be toggled as a favorite.
If it is a favorite, it will be removed.
If it is not a favorite, it will be added.

The response you get will be the updated array of employer_ids, called "favorites".

File: src/api/controllers/users.js [hunk 1]
```diff
@@ -99,4 +99,78 @@ function patchProfileByAuthUser(req, res) {
...
+// student only
+function getFavoritesByAuthUser(req, res) {
+    if (req.user.user_type !== 'student') {
+        return res.status(403).json({
+            "message": response.onlyStudent
+        });
+    } else {
+        try {
+            User.findById(req.user._id).populate('favorites').exec(function(err, user) {
+                return res.status(200).json({
```
Comment (brianwang9100) [inline src/api/controllers/users.js]: is there a benefit of try catch block over doing the if (err) check inside of the exec block?

just curious

File: src/api/controllers/users.js [hunk 2]
```diff
@@ -99,4 +99,78 @@ function patchProfileByAuthUser(req, res) {
...
+// PATCH /users/auth/favorites
+// student only
+// client passes in an objectID of an employer to toggle, receives back the updated list of favorites
+function patchFavoritesByAuthUser(req, res) {
+    if (req.user.user_type !== 'student') {
+        return res.status(403).json({
+            "message": response.onlyStudent
+        });
+    } else {
+        try {
```
Comment (brianwang9100) [inline src/api/controllers/users.js]: same issue as above.

File: src/api/controllers/users.js (hunk 1 above)
Comment (jhayes45) [reply src/api/controllers/users.js]: I don't remember. When I first wrote it I just had an outer try/catch, but that didn't actually catch some rejected promises inside the query block. So then I added the inner try/catch. It is probably best to just keep it consistent with what we had before, which is an inner try/catch block for the inner logic, and and if(err) check to catch outer query errors. 

I will change these to remove the outer try/catch and add if(err) checks

**Pull request merged**